receiver/mysqlreceiver/config_test.go

```diff
@@ -32,6 +32,7 @@ func TestLoadConfig(t *testing.T) {
 	expected.Password = "${env:MYSQL_PASSWORD}"
 	expected.Database = "otel"
 	expected.CollectionInterval = 10 * time.Second
+	expected.TLS.Insecure = true
```

Comment:
Since we have custom unmarshaling logic, I think we need to test it. It should be relatively straightforward to replicate this test and validate against the case where `tls` is included without any customization. If working correctly this should differ from the other config which leaves `tls` undefined. You'll need to add the input [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/mysqlreceiver/testdata/config.yaml). 

```yaml
mysql:
  endpoint: localhost:3306
  username: otel
  password: ${env:MYSQL_PASSWORD}
  database: otel
  collection_interval: 10s
mysql/default_tls:
  endpoint: localhost:3306
  username: otel
  password: ${env:MYSQL_PASSWORD}
  database: otel
  collection_interval: 10s
  tls: # specified, but use default values
```

Reply:
That makes sense, will add that the extra test case